lib/board.rb

```diff
@@ -34,23 +30,30 @@ def valid_placement?(ship, coordinates)
         letter_array = letters.map do |letter|
           letter.ord
         end
+        num_array = numbers.map do |number|
+          number.ord
+        end
+        letter_array.each_cons(2).all? { |x, y| x == y } || letter_array.each_cons(2).all? { |x, y| x == y - 1 }
+        num_array.each_cons(2).all? { |x, y| x == y - 1 }
```

Comment:
Our conceptualization of the horizontal / vertical consecutive cells is coming to fruition